plugins/polio/tasks/api/refresh_lqas_im_data.py

```diff
@@ -0,0 +1,145 @@
+from iaso.api.tasks import ExternalTaskModelViewSet, ExternalTaskPostSerializer, ExternalTaskSerializer, TaskSerializer
+from iaso.models.base import RUNNING, ERRORED, SUCCESS, Task
+from iaso.models.org_unit import OrgUnit
+from rest_framework import permissions, serializers, filters
+from hat.menupermissions import models as permission
+from iaso.api.common import HasPermission
+from rest_framework.response import Response
+from django_filters.rest_framework import DjangoFilterBackend  # type:ignore
+from django.db.models import Q
+from rest_framework.decorators import action
+
+LQAS_TASK_NAME = "Refresh LQAS data"
+LQAS_CONFIG_SLUG = "lqas-pipeline-config"
+IM_TASK_NAME = "Refresh IM data"
+IM_CONFIG_SLUG = "im-pipeline-config"
+NO_AUTHORIZED_COUNTRY_ERROR_MESSAGE = "No authorised org unit found for user"
+NO_AUTHORIZED_COUNTRY_ERROR = {"country_id": NO_AUTHORIZED_COUNTRY_ERROR_MESSAGE}
+NO_AUTHORIZED_COUNTRY_ERROR = {"country_id": "No authorised org unit found for user"}
+
+
+class RefreshLQASIMDataGetSerializer(serializers.Serializer):
+    country_id = serializers.IntegerField(required=False)
+
+    def validate(self, attrs):
+        validated_data = super().validate(attrs)
+        request = self.context["request"]
+        country_id = request.query_params.get("country_id", None)
+        # It seems a bit stange to limit the access on country id
+        # but to launch the refresh for all countries if no id is passed
```

Comment:
I agree with the comment. This lacks a test on the case where the user is limited to some org units and tries to refresh all countries

Reply:
(not a big security risk, on the other hand, refreshes are happening automatically on all countries regularly anyway, this just speeds up the process, so, it's not blocking)

Reply:
We can't refresh all countries from the polio UI, it's just the pipeline code that behaves this way if I recall correctly